earth_enterprise/src/common/khFileUtils.cpp

```diff
@@ -789,9 +791,9 @@ khTmpFilename(const std::string &basename, mode_t mode)
   khEnsureParentDir(fname);
 
   // must make a writable copy for mkstemp to play with
-  char tmpname[fname.size()+7];
-  strcpy(tmpname, fname.c_str());
-  strcat(tmpname, "XXXXXX");
+  int bufsize = fname.size() + 7;
```

Comment:
Did we determine where this magic number is coming from? Might be better to make it a const variable (maybe `uint8_t`?)

Reply:
I'm assuming it's for the 6 Xs and a null term.

Reply:
technically bufsize probably should be a size_t actually.